src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1215,7 +1215,7 @@ public SolutionState AddDocuments(ImmutableArray<DocumentInfo> documentInfos)
             ImmutableArray<DocumentInfo> documentInfos,
             Func<DocumentInfo, ProjectState, T> createDocumentState,
             Func<ProjectState, ImmutableArray<T>, (ProjectState newState, CompilationTranslationAction? translationAction)> addDocumentsToProjectState)
-            where T : TextDocumentState
+            where T : notnull, TextDocumentState
```

Comment:
I added this as it actually did impact the IDE somehow: without it, the compiler in an error scenario with lambda resolution would incorrectly choose T to be TextDocumentState? (and would have an error to that effect) but it'd end up squiggling the entire method and not make it clear where the null came from. I'll remove this but will try to reproduce what caused it since it was quite irritating.